feature-add-meal/src/main/java/com/kinandcarta/lib/add/meal/view/AddMealViews.kt

```diff
@@ -0,0 +1,134 @@
+package com.kinandcarta.lib.add.meal.view
+
+import android.content.Context
+import androidx.compose.material.Text
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Button
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import com.kinandcarta.lib.add.meal.ui.components.AddressField
+import com.kinandcarta.lib.add.meal.ui.components.DateField
+import com.kinandcarta.lib.add.meal.ui.components.InputTextField
+import com.kinandcarta.lib.add.meal.ui.components.TemperatureField
+import com.kinandcarta.lib.add.meal.viewmodel.AddMealViewModel
+
+@Composable
+internal fun AddMealScreen(addMealViewModel: AddMealViewModel, context: Context) {
+    val previousState = remember { mutableStateOf(addMealViewModel.state) }
+    val showDialog = remember { mutableStateOf(true) }
+
+    when (addMealViewModel.state) {
+        is AddMealViewModel.State.Initial -> {
+            LoadedView(addMealViewModel = addMealViewModel, context = context)
+        }
+        is AddMealViewModel.State.Loading -> {
+            LoadingView()
+        }
+        is AddMealViewModel.State.Success -> {
+            if (previousState.value != AddMealViewModel.State.Success) {
```

Comment:
Only show the alert dialog when the state just changed and reset the showDialog value to true in order to present a dialog.